commands/timeTable.js

```diff
@@ -1,156 +1,161 @@
-                option
-                    .setName('分野')
-                    .setDescription('分野を指定します')
-                    .setRequired(true)
-                    .addChoices(
-                        { name: 'M-機械工学分野', value: 'M' },
-                        { name: 'E-電気工学分野', value: 'E' },
-                        { name: 'D-電子工学分野', value: 'D' },
-                        { name: 'J-情報工学分野', value: 'J' },
-                        { name: 'A-建築学分野', value: 'A' },
-                    )
-            )
-            .addStringOption(option =>
-                option
-                    .setName('曜日')
-                    .setDescription('曜日を指定します。指定なければ次の学校の日になります。')
-                    .setRequired(false)
-                    .addChoices(
-                        { name: '月曜日', value: '1' },
-                        { name: '火曜日', value: '2' },
-                        { name: '水曜日', value: '3' },
-                        { name: '木曜日', value: '4' },
-                        { name: '金曜日', value: '5' },
-                    )
-            ),
+  {
+    data: new SlashCommandBuilder()
+      .setName("timetable")
+      .setDescription("指定した分野・曜日の時間割を送信します")
+      .addStringOption((option) =>
+        option
+          .setName("分野")
+          .setDescription("分野を指定します")
+          .setRequired(true)
+          .addChoices(
+            { name: "M-機械工学分野", value: "M" },
+            { name: "E-電気工学分野", value: "E" },
+            { name: "D-電子工学分野", value: "D" },
+            { name: "J-情報工学分野", value: "J" },
+            { name: "A-建築学分野", value: "A" }
+          )
+      )
+      .addStringOption((option) =>
+        option
+          .setName("曜日")
+          .setDescription(
+            "曜日を指定します。指定なければ次の学校の日になります。"
+          )
+          .setRequired(false)
+          .addChoices(
+            { name: "月曜日", value: "1" },
+            { name: "火曜日", value: "2" },
+            { name: "水曜日", value: "3" },
+            { name: "木曜日", value: "4" },
+            { name: "金曜日", value: "5" }
+          )
+      ),
 
-        async execute(interaction) {
+    async execute(interaction) {
+      // get strings from options 分野
+      const bunya = interaction.options.getString("分野");
+      // get strings from options 曜日
+      let dt = new Date();
+      let dayOfWeek = dt.getDay();
+      let hours = dt.getHours();
+      if (interaction.options.getString("曜日") === "1") {
+        dayOfWeek = 1;
+      } else if (interaction.options.getString("曜日") === "2") {
+        dayOfWeek = 2;
+      } else if (interaction.options.getString("曜日") === "3") {
+        dayOfWeek = 3;
+      } else if (interaction.options.getString("曜日") === "4") {
+        dayOfWeek = 4;
+      } else if (interaction.options.getString("曜日") === "5") {
+        dayOfWeek = 5;
+      } else {
+        if (hours >= 17) {
+          dayOfWeek += 1;
+        }
+        if (dayOfWeek === 6 || dayOfWeek === 7 || dayOfWeek === 0) {
+          dayOfWeek = 1;
+        }
+      }
```

Comment:
stringからnumberに変換したらきれいなコードになるかも
とか思ったけど、例えば曜日に `2serhgre` って打ったら `2` になるからどうなんだろう

```suggestion
let parsedDayOfWeek = parseInt(interaction.options.getString("曜日"));
let dayOfWeek = dt.getDay();
if (!isNaN(parsedDayOfWeek) && dayOfWeek >= 1 && dayOfWeek <= 5)) {
  dayOfWeek = parsedDayOfWeek;
} else {
  if (hours >= 17) dayOfWeek++;
  if (dayOfWeek === 6 || dayOfWeek === 7 || dayOfWeek === 0) dayOfWeek = 1;
}

Reply:
なんで室蘭にしてたんやろ…

Reply:
```suggestion
      let parsedDayOfWeek = parseInt(interaction.options.getString("曜日"));
      let dayOfWeek = dt.getDay();
      if (!isNaN(parsedDayOfWeek) && dayOfWeek >= 1 && dayOfWeek <= 5) {
        dayOfWeek = parsedDayOfWeek;
      } else {
        if (hours >= 17) {
          dayOfWeek += 1;
        }
        if (dayOfWeek === 6 || dayOfWeek === 7 || dayOfWeek === 0) {
          dayOfWeek = 1;
        }
      }
```

コードミスを修正してこんなんでどうでしょ

Reply:
いいと思います👍